wire-ios-sync-engine/Source/Synchronization/EventProcessor.swift

```diff
@@ -225,3 +261,32 @@ extension Notification.Name {
     /// Published after the last event has been processed.
     static let eventProcessorDidFinishProcessingEventsNotification = Self("EventProcessorDidFinishProcessingEvents")
 }
+
+private actor ProcessedEventList {
+
+    private var hashes = Set<Int64>()
+
+    // A full list would contain approx 80kB.
+    private let capacity = 10_000
+
+    func addEvent(_ event: ZMUpdateEvent) {
+        guard let hash = event.contentHash else {
+            return
```

Comment:
The `StoreUpdateEvent`s should all a `contentHash` right and therefore this shouldn't happen? Or am I missing something?

If yes then at a minimum I would assert and log an error here.

Is it expected that `event.contentHash` might be nil? I guess initially after the app updates? I'm wondering if

Reply:
yes that's correct, I'll add the assertion

Reply:
done 75842748fdfcc876752f5fa3183f1cd7c8bc095d